### PR TITLE
MM-19308 Fixed date wrapping

### PR DESCRIPTION
**Pull request by imisshtml**
#### Summary
Updated the displayName to check for the reply button and bot tag. If seen the displayName container is set from 60% to 55% to allow for the extra space.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-19308

#### Checklist
- [ ] Has UI changes - displayName container width goes from 60% to 55%

#### Device Information
This PR was tested on: 
iPhone X 12.4 (Simulator)
iPhone Xs Max 12.3

#### Screenshots
Simulator Screen Shot - iPhone X - 2019-10-18 at 22 28 41

### CODE REVIEW

File: app/components/post_header/post_header.js [hunk 1]
```diff
@@ -152,7 +158,7 @@ export default class PostHeader extends PureComponent {
             return (
                 <TouchableWithFeedback
                     onPress={this.handleUsernamePress}
-                    style={style.displayNameContainer}
+                    style={[style.displayNameContainer, {maxWidth: showReply && isBot ? '55%' : '60%'}]}
```
Comment (enahum) [inline app/components/post_header/post_header.js]: the `displayNameContainer` style already defines `maxWidth` as 60%, in order to av oid unneeded re-renders by changing the style object every time, you could define a second style like
```
displayNameContainerBotReplyWidth: {
    maxWidth:  '55%',
},
```

the do something like 
```
const displayNameStyle =  [style.displayNameContainer, showReply && isBot ? style.displayNameContainerBotReplyWidth : null];
```

and in the `TouchableWithFeedback style={displayNameStyle}` or something like that

Comment (enahum) [reply app/components/post_header/post_header.js]: and definitely write some unit tests for this

Comment (imisshtml) [reply app/components/post_header/post_header.js]: Updated

**Comment by imisshtml** [comment]
> Tested and looks good, except for one: on landscape, the display name is truncated even though there is enough space to fully render it (same case when viewed as flagged post).
> Screen Shot 2019-10-22 at 5 53 48 PM

Hey @saturninoabril, this was the current functionality beforehand. Should I update this item to adjust the landscape also? Here is a screenshot of landscape without my changes:

Simulator Screen Shot - iPhone X - 2019-10-24 at 22 28 39

**Comment by saturninoabril** [comment]
@imisshtml Thanks for checking.  I filed a separate ticket to address it - https://mattermost.atlassian.net/browse/MM-19708.  I'm good as is.

File: app/components/post_header/post_header.js [hunk 2]
```diff
@@ -117,9 +117,16 @@ export default class PostHeader extends PureComponent {
...
+            renderReplies,
+            shouldRenderReplyButton,
+            commentedOnDisplayName,
+            commentCount,
+            isBot,
         } = this.props;
 
         const style = getStyleSheet(theme);
+        const showReply = shouldRenderReplyButton || (!commentedOnDisplayName && commentCount > 0 && renderReplies);
+        const displayNameStyle = [style.displayNameContainer, showReply && isBot ? style.displayNameContainerBotReplyWidth : null];
```
Comment (enahum) [inline app/components/post_header/post_header.js]: I believe this also applies (the bot tag) when the message is from a webhook or an autoresponder not only when the post is by a bot user

Comment (imisshtml) [reply app/components/post_header/post_header.js]: Got it, I should have updated that area also, on it!

**Comment by hmhealey** [comment]
Closing and reopening the PR to re-trigger the build

**Comment by enahum** [comment]
Build is passing but with the previous job.

**Pull request merged**